StepsToCare/CareDashboard.ascx.cs

```diff
@@ -1072,9 +1124,25 @@ public void gList_RowDataBound( object sender, System.Web.UI.WebControls.GridVie
                         {
                             careNeedFlagStr = "<i class=\"fas fa-flag text-danger mx-2\"  data-toggle=\"tooltip\" title=\"Not enough Care Touches\"></i>";
                         }
+                        else if ( hasTouchTemplates )
+                        {
+                            var catTouchTemplates = _touchTemplates.GetValueOrDefault( careNeed.CategoryValueId.Value, new List<TouchTemplate>() );
+                            foreach ( var template in catTouchTemplates )
+                            {
+                                var templateNotes = careNeedNotesList.Where( n => n.Text.Equals( template.NoteTemplate.Note ) );
+                                var templateNotesToCheckInHours = templateNotes.Where( n => ( RockDateTime.Now - n.CreatedDateTime ).Value.TotalHours <= template.MinimumCareTouchHours );
+                                var templateFlag = ( ( template.Recurring && templateNotesToCheckInHours.Count() < template.MinimumCareTouches ) || ( !template.Recurring && templateNotes.Count() < template.MinimumCareTouches ) ) && dateDifference.TotalHours >= template.MinimumCareTouchHours;
+
+                                if ( templateFlag )
+                                {
+                                    careNeedFlagStr = $"<i class=\"fas fa-flag text-warning mx-2\"  data-toggle=\"tooltip\" title=\"Not enough '{template.NoteTemplate.Note}' Care Touches!\"></i>";
```

Comment:
Move this to a variable building the tooltip and then attach the flag outside of the foreach loop.